### ISSUE TITLE
Data retrieving time using JSON APIs

### ISSUE THREAD

**Issue by sajithaliyanage** [moneroexamples/onion-monero-blockchain-explorer#142]
If we used `http://0.0.0.0:8081/api/block/` API to retrieve blocks, it gives different time durations to finish.
I used this API to get data of block ranges. **One range consists of 100,000 blocks**. So 1-1,000,000 blocks(10 iterations) each range took an average time of **500seconds**. But after 1 million blocks other ranges took an average time of **1300seconds**. 

Is there any reason for this? I think retrieving data from after 1000,000 blocks are getting slow.

**Comment by moneroexamples** [comment]
You mean this api (https://github.com/moneroexamples/onion-monero-blockchain-explorer/tree/devel#apiblockblock_numberblock_hash)? It only takes block number or hash?

But anyway, if you want to fetch so many blocks, why just not to use monero deamon's api directly to get 
blocks from the blockchain?

**Comment by sajithaliyanage** [comment]
Yes, I used the above-mentioned API to get block data. I used this API inside a for loop to pass the block height as the iterator. The issue of this is after 1 million blocks this retrieving process getting slower.

**Block Range**     ->   Retrieivng Time(seconds)
**1-100,000**                        ->945.87
**100000 - 199999**             ->503.88
**200000 - 299999**             ->681.37
.......
**900000 - 999999**             ->403.32
_**1000000 - 1099999      ->1623.32  ( 3 times bigger than previous ranges)
1100000 - 1199999         ->1589.22 ( 3 times bigger than previous ranges)**_

Any reason for this? Or is this problem related to Monero Daemon?

**Comment by moneroexamples** [comment]
I think that around this there was hard fork (v2, block 1009827) to force all txs to have ring size >=3, subsequently making txs bigger. Also in that hard fork block time was increased to 2 minutes, making blocks also bigger. Add to this greater interest in monero (more txs being made) and this could possibly explain slower times observed.
